### DISCUSSION TITLE
How to Enable/Inject Trace ID and Span ID for all the Application Logs through logrus

### DISCUSSION THREAD

**Discussion by arjunkanti** [open-telemetry/opentelemetry-go#2988] [Q&A]
Hi,

I am trying to enable Trace ID and Span ID for every application log. I am using opentelemtry APIs to inject and logrus labrary for logging. Below link has example code where i am able to enable trace/span entries to all logs but the trace and span id values are always 0's. Log entry Context doesn't have the http request context trace/span IDs.

Is there any way to get it automatically without injecting context to all the log entries..??

**More Details :** https://stackoverflow.com/questions/72812236/golang-logrus-enable-opentelemetry-trace-id-and-span-id-in-all-application-logs

**Comment by dmathieu** [comment]
No.
The only way to retrieve the current span in Go is through the context. So you do need to have that available for each log entry.

**Reply by arjunkanti** [reply]
Ok.. Thanks...

What about logrus entry.Context..? I have implemented logrus.Format function to get context and append trace id and span id to each logs. But logs always has 0 values.. 

```
func (l customLogger) Format(entry *log.Entry) ([]byte, error) {
        span := trace.SpanFromContext(entry.Context)
        entry.Data["trace_id"] = span.SpanContext().TraceID().String()
        entry.Data["span_id"] = span.SpanContext().SpanID().String()
        return l.formatter.Format(entry)
}
```

Full Example : https://stackoverflow.com/questions/72812236/golang-logrus-enable-opentelemetry-trace-id-and-span-id-in-all-application-logs

**Comment by arjunkanti** [comment]
Is there any plans to enable automatic injection to logs in future releases..?? Without Passing request context to each log entry.

**Reply by Aneurysm9** [reply]
> Is there any plans to enable automatic injection to logs in future releases..?? Without Passing request context to each log entry.

No.  Go requires explicit context propagation.  I don't foresee that changing any time soon.
